umh-core/pkg/service/s6/s6.go

```diff
@@ -315,52 +315,80 @@ func (s *DefaultService) createS6ConfigFiles(ctx context.Context, servicePath st
+//   - <logBase>/<name>          (rotated log directory)
+//
+// The method is called once per reconcile *tick* while the S6-FSM is in the
+// *removing* state, therefore it must:
+//
+//  1. **Return quickly** – never wait or poll.
+//  2. **Be idempotent**  – safe to call when directories are half-gone.
+//  3. **Return nil only when nothing is left.**
+//
+// Any remaining file or I/O error leads to a non-nil return so the FSM keeps
+// trying (or escalates after the back-off threshold).
 func (s *DefaultService) Remove(ctx context.Context, servicePath string, fsService filesystem.Service) error {
 	start := time.Now()
 	defer func() {
-		metrics.ObserveReconcileTime(metrics.ComponentS6Service, servicePath+".remove", time.Since(start))
+		metrics.ObserveReconcileTime(metrics.ComponentS6Service,
+			servicePath+".remove", time.Since(start))
 	}()
 
-	if s.logger != nil {
-		s.logger.Debugf("Removing S6 service %s", servicePath)
-	}
-
-	exists, err := s.ServiceExists(ctx, servicePath, fsService)
-	if err != nil {
-		return fmt.Errorf("failed to check if service exists: %w", err)
-	}
-	if !exists {
-		return ErrServiceNotExist
+	if ctx.Err() != nil {
+		return ctx.Err() // context already cancelled / deadline exceeded
 	}
 
-	// Remove the service from contents.d first
-	serviceName := filepath.Base(servicePath)
-	//contentsFile := filepath.Join(filepath.Dir(servicePath), "user", "contents.d", serviceName)
-	//fsService.Remove(ctx, contentsFile) // Ignore errors - file might not exist
+	srvName := filepath.Base(servicePath)
+	logDir := filepath.Join(constants.S6LogBaseDir, srvName)
 
-	// Remove the service directory
-	err = fsService.RemoveAll(ctx, servicePath)
-	if err != nil {
-		return fmt.Errorf("failed to remove S6 service %s: %w", servicePath, err)
+	// ──────────────────────── fast-path ────────────────────────────
+	// If both paths are already gone we are done – lets the FSM fire
+	// "remove_done" immediately and exit the reconcile loop.
+	if gone, _ := fsService.PathExists(ctx, servicePath); !gone {
+		if gone2, _ := fsService.PathExists(ctx, logDir); !gone2 {
+			return nil
+		}
 	}
 
-	// Clean up logs directory (best effort - don't block removal if this fails)
-	logDir := filepath.Join(constants.S6LogBaseDir, serviceName)
-	if logErr := fsService.RemoveAll(ctx, logDir); logErr != nil && s.logger != nil {
-		sentry.ReportServiceErrorf(
-			s.logger,
-			serviceName,
-			"s6",
-			"cleanup_logs",
-			"Failed to clean up log directory: %v",
-			logErr,
-		)
-	}
+	//----------------------------------------------------------------
+	// Two independent best-effort deletions.  We *always* call them,
+	// even if the directory is missing, because RemoveAll on a non-
+	// existent path is cheap and returns nil – keeps the code simple.
+	//----------------------------------------------------------------
+	srvErr := fsService.RemoveAll(ctx, servicePath)
+	logErr := fsService.RemoveAll(ctx, logDir)
+
```

Comment:
Why is there no fast abort here, if any of these error ?

Reply:
Because the force remove is only executed once. So we want to remove as much as possible even when errrors occur.